packages/cadl-ranch-specs/http/models/enum-properties/main.cadl

```diff
@@ -10,7 +10,7 @@ using Azure.Core.Foundations;
 @server("http://localhost:3000", "TestServer endpoint")
 @doc("Illustrates input, output, and round-trip models with required enum type properties.")
 @route("/enum-properties-basic")
-namespace Cadl.TestServer.EnumPropertiesBasic;
+namespace EnumPropertiesBasic;
```

Comment:
Wonder if all those scenarios that are related to models. Should maybe be `Model.XYZ` so `Model.EnumProperties`. This would help keeping a good structure in the coverage dashboard where we can group by area well.

 Does basic here also add any value? We are just testing a model with enum properties works fine.

Reply:
Second the `Model` namespace, esp for the dashboard. With that we can also feel more comfortable flattening out the model test folder

Reply:
I think `model.xxx` is good idea for a clear coverage dashboard and flatting test folder. 
About `Does basic here also add any value? We are just testing a model with enum properties works fine.` `extensible-enums` mainly test `@knownValues` and `models/enum-properties` test both basic enum and `@knownValues`

Reply:
I think the difference between those tests is more:
- `extensible-enums` tests enums as a standalone entity
- `models.enumproperties` tests enums as a property of models

I think generally I would recommend tests like these be flattened, but we are also placing a special emphasis on models (because that's what elevates cadl from dpg, especially for our first sprint). So I think it's still valid to keep these separate (at least for now), and make clear the purpose of each

Reply:
So this test test that a model can have property of type enum or extensible enum. It shouldn't actually be testing teh enum and extensible enum too much, just that they work with model. That job should be left to an Enum and `ExtensibleEnum` set of scenarios that are not tied to model.

I think this is the same for a lot of the other model tests. If the goal is really just to check that you can use type X or that it works with optional properties, etc. on model property then it should limit the test to that. If the goal is to test the type X in depth then it should have its own dedicated spec outside of the model scope.

Reply:
Yeah agreed, and that has me thinking: Do we want to have dedicated tests (i.e. a dedicated test for enums, and a dedicated test for dictionaries), and just one model cadl that tests all of these as properties (i.e. this model has an enum and a dictionary property). I'm just worried we're going to balloon with all of these tiny cadl files that could easily be one cadl file

Reply:
too many tiny cadl files will make test troublesome while a big cadl that contains many kinds of property may make test too complicated. I think we could use eclectic way: one cadl shall test for 3~5 kinds of property. It takes both flexibility and complexity into account.

Reply:
talked with @iscai-msft last night we believe we'll structure things this way:
- one spec for model property types. Each type that we want to test gets a `get` and a `put`. This WILL NOT be testing the types in depth. Just make sure they work fine as model properties.
- one spec for optional model properties
- one spec for nullable properties
- one spec for model visibility(read only write only)
- one spec per different entity type(enum, union, array, etc.) where we test that entity in depth but not with models